src/gui/mainwindow.cpp

```diff
@@ -452,18 +453,21 @@ MainWindow::MainWindow(QWidget *parent)
 
     qDebug("GUI Built");
 #ifdef Q_OS_WIN
-    if (!pref->neverCheckFileAssoc() && (!Preferences::isTorrentFileAssocSet() || !Preferences::isMagnetLinkAssocSet())) {
-        if (QMessageBox::question(this, tr("Torrent file association"),
-                                  tr("qBittorrent is not the default application to open torrent files or Magnet links.\nDo you want to associate qBittorrent to torrent files and Magnet links?"),
-                                  QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes) == QMessageBox::Yes) {
-            Preferences::setTorrentFileAssoc(true);
-            Preferences::setMagnetLinkAssoc(true);
-        }
-        else {
-            pref->setNeverCheckFileAssoc();
-        }
-    }
-#endif
+    if (!pref->neverCheckFileAssoc() && !Utils::OS::isAssociationsSet()) {
+        const QString dlgTitle {tr("Torrent file association")};
+        const QString dlgText {((QSysInfo::WindowsVersion <= QSysInfo::WV_WINDOWS7)
+                    ? tr("qBittorrent is not the default application to open torrent files or magnet links.\n"
+                         "Do you want to associate qBittorrent with torrent files and magnet links?")
+                    : tr("Do you want to set qBittorrent as the default program for torrent files and magnet links?\n"
+                         "You can do it later using \"Default Programs\" dialog from the Control Panel."))};
+
+        int dialogResult = QMessageBox::question(this, dlgTitle, dlgText, QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes);
+        if (dialogResult == QMessageBox::Yes)
+            Utils::OS::setAssociations();
+        if ((dialogResult != QMessageBox::Yes) || (QSysInfo::WindowsVersion >= QSysInfo::WV_WINDOWS8))
```

Comment:
I find it confusing why there is a check for `ver >= win8`??

Reply:
Win8+ can't actually check whether it set as default program or not. So we just ask the user itself to set it on the first run and we never do it again.